V3.0/UFTP-V3.0-metering.xsd

```diff
@@ -0,0 +1,157 @@
+                </xs:annotation>
+            </xs:enumeration>
+        </xs:restriction>
+    </xs:simpleType>
+    <!-- -->
+    <!--  Metering Profile Enum -->
+    <!-- -->
+    <xs:simpleType name="MeteringProfileEnum">
+        <xs:restriction base="xs:string">
+            <xs:enumeration value="Power">
+                <xs:annotation>
+                    <xs:documentation>The average active power during ISP, considering both import and export energy. Power=(ImportEnergy-ExportEnergy)*2
+                        Including the power profile is recommended. It is expected that in the following major version the power will become a mandatory value.
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="ImportEnergy">
+                <xs:annotation>
+                    <xs:documentation>Imported active energy, consumed during the ISP</xs:documentation>
+                </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="ExportEnergy">
+                <xs:annotation>
+                    <xs:documentation>Exported active energy, generated during the ISP</xs:documentation>
+                </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="ImportMeterReading">
+            <xs:annotation>
+                <xs:documentation>Cumulative metered imported active energy reading, at the end of the ISP</xs:documentation>
+            </xs:annotation>
+            </xs:enumeration>
+            <xs:enumeration value="ExportMeterReading">
+                <xs:annotation>
+                    <xs:documentation>Cumulative metered exported active energy reading, at the end of the ISP</xs:documentation>
+                </xs:annotation>
+            </xs:enumeration>
+        </xs:restriction>
+    </xs:simpleType>
+    <!-- -->
+    <!--  Metering Message Type -->
+    <!-- -->
+    <xs:complexType name="MeteringMessageType">
+        <xs:complexContent>
+            <xs:extension base="PayloadMessageType">
+                <xs:sequence>
+                    <xs:annotation>
+                        <xs:documentation>The message must contain one or more profiles with metering, energy or price values.</xs:documentation>
+                    </xs:annotation>
+                    <xs:element name="Profile" type="MeteringProfileType" maxOccurs="unbounded"/>
+                </xs:sequence>
+                <xs:attribute name="Revision" type="xs:long" use="required">
+                    <xs:annotation>
+                        <xs:documentation>Revision of this message. A sequence number that must be incremented each time a new revision of a metering message is sent.</xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:attribute name="ISP-Duration" type="xs:duration" use="required">
```

Comment:
Same, is my comment on the TimeZone attribute, this should be part of PayloadMessageType.
I suggest you move this attribute to PayloadMessageType